### PR TITLE
#160452489 List users functionality

### PR DESCRIPTION
**Pull request by Alameen688**
#### What does this PR do?
Implements list users (authors) functionality by adding an endpoint to get a list of authors and also an endpoint to get author's profile by id
#### Description of Task to be completed?
- Add endpoint for list users 
- Support pagination
- Refractor existing endpoint for getting logged in user's profile to also get a user's profile by id
#### How should this be manually tested?
- Run `git fetch` to fetch new branch
- Checkout to this branch `git checkout ft-160452489-list-users`
- Run `npm install` to install dependencies
- Run `npm start` to start application
- Run `npm test` to run tests
- [see docs](http://localhost:3000/api-docs/)
[GET](http://localhost:3000/api-docs/#/Users/get_users_list) `/api/v1/users/list?page=1&size=10`
[GET](http://localhost:3000/api-docs/#/Users/get_users__id_) `/api/v1/users/{id}`
#### What are the relevant pivotal tracker stories?
[#160452489](https://www.pivotaltracker.com/story/show/160452489)
#### Screenshots (if appropriate)
- N/A
#### Questions:
- N/A

### CODE REVIEW

File: controllers/user/UserProfile.js [hunk 1]
```diff
@@ -47,38 +46,74 @@ class UserProfile {
...
+   *
+   *
+   * @static
+   * @param {request} req
+   * @param {response} res
+   * @param {function} next
+   * @returns {object} return a json response
+   * @memberof UserProfile
+   */
+  static async getAllProfile(req, res, next) {
```
Comment (macphilips) [inline controllers/user/UserProfile.js]: @Alameen688 There is an error with the pagination you implemented. In your `User.findAll` you are excluding the currently logged in user, but `User.count()` will include that user in the count. This makes the pagination inaccurate. View screenshot
screen shot 2018-10-11 at 4 17 17 pm

From the screenshot, if the total number of users is 5 when page=3 and size=2, the user should get one result not an empty array.

File: swagger.yaml [hunk 2]
```diff
@@ -631,6 +631,65 @@ paths:
...
+      summary: Get a list of authors
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: "page"
+          in: "query"
+          description: "Page number"
+          required: true
```
Comment (macphilips) [inline swagger.yaml]: Please make the query parameters optional.

**Review by macphilips — CHANGES_REQUESTED** [review]

File: controllers/user/UserProfile.js (hunk 1 above)
Comment (Alameen688) [reply controllers/user/UserProfile.js]: Thanks for spotting that. @macphilips. Corrected!

File: swagger.yaml (hunk 2 above)
Comment (Alameen688) [reply swagger.yaml]: It's actually optional in the implementation, I missed it on the swagger docs. I'll correct that now. Thanks

**Review by hnobi — APPROVED** [review]

**Review by macphilips — APPROVED** [review]
Great job maintaining the test coverage. Everything looks good to me. This pull request meets the required standards of the team and meets the acceptance criteria of the associated story. PR approved. Cheers

**Pull request merged**